### ISSUE TITLE
Issue with adding meterial-ui library

### ISSUE THREAD

**Issue by wer32** [mxstbr/react-boilerplate#1202]
Hi,

Have anybody successfully added material-ui library to react-boilerplate for static resources?

I have some issues with adding material-ui to react-boilerplate.
The issue is only with static resources. I mean if run **npm run build** and then put all resources from /build folder to web server then project will not work correct. 
But it works well when start project **npm run start**.

Steps to reproduce:
1. add to package.json
`"material-ui": "0.16.1"`

2. add to package.json
`"react-tap-event-plugin": "1.0.0",`

3. replace 
```
const render = (messages) => {
  ReactDOM.render(
    <Provider store={store}>
      <LanguageProvider messages={messages}>
        <Router
          history={history}
          routes={rootRoute}
          render={
            // Scroll to top when going to a new page, imitating default browser
            // behaviour
            applyRouterMiddleware(useScroll())
          }
        />
      </LanguageProvider>
    </Provider>,
    document.getElementById('app')
  );
};
``` 
with next code
```
const injectTapEventPlugin = require('react-tap-event-plugin');
injectTapEventPlugin();

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const render = (messages) => {
  ReactDOM.render(
    <Provider store={store}>
      <MuiThemeProvider>
        <LanguageProvider messages={messages}>
          <Router
            history={history}
            routes={rootRoute}
            render={
              // Scroll to top when going to a new page, imitating default browser
              // behaviour
              applyRouterMiddleware(useScroll())
            }
          />
        </LanguageProvider>
      </MuiThemeProvider>
    </Provider>,
    document.getElementById('app')
  );
};
```

4. add 
`import AppBar from 'material-ui/AppBar';`

5. add simple material-ui component to app/containers/App/index.js
```
function App(props) {
  return (
    <AppWrapper>
      <Helmet
        titleTemplate="%s - React.js Boilerplate"
        defaultTitle="React.js Boilerplate"
        meta={[
          { name: 'description', content: 'A React.js Boilerplate application' },
        ]}
      />
      <Header />
      <AppBar
        title="Title"
        iconClassNameRight="muidocs-icon-navigation-expand-more"
      />
      {React.Children.toArray(props.children)}
      <Footer />
    </AppWrapper>
  );
}
```

After build project and deploy to webserver any browser show next exception in cosole

```
main.54bb541….js:1 Uncaught TypeError: Cannot read property 'document' of undefined
    at Object../ (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:16840)
    at Object.<anonymous> (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:1892)
    at t (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:101)
    at Object../ (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:16836)
    at Object.<anonymous> (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:1892)
    at t (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:101)
    at Object../node_modules/core-js/modules/_export.js (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:8:6258)
    at t (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:101)
    at Object../ (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:16836)
    at Object.<anonymous> (http://localhost:8080/main.54bb541f4c7e1f0806bc.js:1:1892)
```

**Comment by samit4me** [comment]
Can confirm the same issue, I've tried a few different things, like using different material ui component or moving or only using the `<MuiThemeProvider>` and always get the same error. My first thoughts were if material ui was using `context` this may not work with some of the library's in this boilerplate, `react-redux` springs to mind. Unfortunately I don't have time to investigate right now and I was not able to find reports of this issue elsewhere. So it's tagged as unconfirmed until the root cause(s) can be found!

**Comment by wer32** [comment]
@samit4me Unfortunatelly I don't have profound knoledge in node.js but if can give me direction or instruction how can I found the root of cause(s) I can spend some time.

**Comment by samit4me** [comment]
Okay, so I added `material-ui` to a [react-gif-search app](https://github.com/samit4me/react-gif-search) I was toying around with, which is based on `create-react-app` but has Redux, react-router and I added react-intl to more closely match this repo.

Guess what it worked 😄

---

Knowing it wasn't react-redux or react-intl causing the issues I checked out the `dev` branch of the repo and did the following:
- `npm i -S react-tap-event-plugin matrial-ui`
- In `app.js` I replaced the `render()` function with:
```
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import injectTapEventPlugin from 'react-tap-event-plugin';
injectTapEventPlugin();

const render = (messages) => {
  ReactDOM.render(
    <MuiThemeProvider>
      <Provider store={store}>
        <LanguageProvider messages={messages}>
          <Router
            history={history}
            routes={rootRoute}
            render={
              // Scroll to top when going to a new page, imitating default browser
              // behaviour
              applyRouterMiddleware(useScroll())
            }
          />
        </LanguageProvider>
      </Provider>
    </MuiThemeProvider>,
    document.getElementById('app')
  );
};
```
- Then added these two lines into `HomePage/index.js` in the appropiate places:
```
import RaisedButton from 'material-ui/RaisedButton';
...
<RaisedButton label="Default" />
```

You won't believe it, but it rendered the material-ui button 😄 🎉

---

In summary, I have no idea why it is NOT working in `master` but there has been 10 PRs merged into `dev` since the last release. My suggestion would be checkout the `dev` branch and see how you go, good news is that is should be fixed in the next release. Hope this helps, let us know how you go.

**Comment by futzco** [comment]
Hi, this is what works for me:

In `app.js` add (for my sanity I added the below lines just after the `// Import all the third party stuff` section):

```
import injectTapEventPlugin from 'react-tap-event-plugin';
injectTapEventPlugin();
```

Then in `containers/App/index.js` add the `<MuiThemeProvider>` just after the `return (`. Mine looks like this (probably changed it a little along the way other than adding the `<MuiThemeProvider>`, but you'll get the idea):

```
function App(props) {
  return (
    <MuiThemeProvider>
      <section className={styles.wrapper}>
        <Main>
          {React.Children.toArray(props.children) }
        </Main>
        <Footer />
      </section>
    </MuiThemeProvider>
  );
}
```

For the sake of this example, if your `<Footer />` component invoked above looked exactly like this, it would work like a charm:

```
import React from 'react';

import RaisedButton from 'material-ui/RaisedButton';

import styles from './styles.css';

function Footer() {
  return (
    <footer className={styles.outer}>
      <RaisedButton
        className={styles.button}
        disabled={false}
        fullWidth={true}
        keyboardFocused={true}
        label="Fuck! It really works."
        labelPosition="after"
        primary={true}
        secondary={false}
      />
    </footer>
  );
}

export default Footer;
```

Hopefully this is a good start. Good luck!

**Comment by wer32** [comment]
@futzco have you tried to run this with static generated resources from /build folder?
I've tried to follow your instruction and it works pefrectly if you run npm run start
but still get the same issue when run these static files from apachy server.

**Comment by wer32** [comment]
@samit4me
I tried dev branch and copy all changes exactly as you said. But it still now works if I put generated static content from /build folder in apache server. But works as expected for mnp run start.
That's really confusing.

```
main.b2108cb….js:1 Uncaught TypeError: Cannot read property 'document' of undefined
    at Object../ (http://localhost:8080/main.b2108cb64429ca259424.js:1:16840)
    at Object.<anonymous> (http://localhost:8080/main.b2108cb64429ca259424.js:1:1892)
    at t (http://localhost:8080/main.b2108cb64429ca259424.js:1:101)
    at Object../ (http://localhost:8080/main.b2108cb64429ca259424.js:1:16836)
    at Object.<anonymous> (http://localhost:8080/main.b2108cb64429ca259424.js:1:1892)
    at t (http://localhost:8080/main.b2108cb64429ca259424.js:1:101)
    at Object../node_modules/core-js/modules/_export.js (http://localhost:8080/main.b2108cb64429ca259424.js:8:5822)
    at t (http://localhost:8080/main.b2108cb64429ca259424.js:1:101)
    at Object../ (http://localhost:8080/main.b2108cb64429ca259424.js:1:16836)
    at Object.<anonymous> (http://localhost:8080/main.b2108cb64429ca259424.js:1:1892)
```

**Comment by anhldbk** [comment]
Here you go: http://stackoverflow.com/questions/37225592/react-router-withrouter-does-not-inject-router
